pkg/antctl/command_definition.go

```diff
@@ -320,6 +327,209 @@ func respTransformer(obj interface{}) (interface{}, error) {
 	return target, nil
 }
 
+// getTableHeaders get the headers of the output table.
+func (cd *commandDefinition) getTableHeaders(target interface{}, tab *table.Writer) ([]string, error) {
+	args := make([]string, 0)
+	headers, ok := target.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("converting types error in getTableHeaders")
+	}
+
+	// ensure that "name" column is the first column if it exists
+	// sort other columns in alphabet order in default
+	hasNameHeader := false
+	nameHeader := ""
+	tabConfigs := make([]table.ColumnConfig, 0)
+	for header := range headers {
+		if strings.ToLower(header) == "name" {
+			hasNameHeader = true
+			nameHeader = header
+			continue
+		}
+		args = append(args, header)
+	}
+	sort.Strings(args)
+	if hasNameHeader {
+		args = append([]string{nameHeader}, args...)
+	}
+
+	tabHeaders := table.Row{}
+	// columnNameMap stores column names that are needed to be converted in text
+	columnNameMap := map[string]string{
+		"appliedtogroups": "applied-to",
+	}
+	// set the column configs
+	for _, arg := range args {
+		var colName string
+		if val, ok := columnNameMap[strings.ToLower(arg)]; ok {
+			colName = val
+		} else {
+			colName = strings.ToUpper(arg)
+		}
+		tabHeaders = append(tabHeaders, colName)
+		tabConfig := table.ColumnConfig{
+			Name:     colName,
+			WidthMax: maxTableOutputColumnLength,
+		}
+		tabConfigs = append(tabConfigs, tabConfig)
+	}
+	(*tab).AppendHeader(tabHeaders)
+	(*tab).SetColumnConfigs(tabConfigs)
+
+	return args, nil
+}
+
+func (cd *commandDefinition) handlePodsName(pods []interface{}) (string, error) {
```

Comment:
Can we have this specific conversion function in the command's package itself? I think the file is supposed to include generic code only.

Reply:
> Can we have this specific conversion function in the command's package itself? I think the file is supposed to include generic code only.

You mean that move this part to the corresponding file in folder apiserver? I am afraid I have to change the struct of responses in this way. Is it ok to do so?

Reply:
I meant moving them to pkg/antctl/transform/xxxx/, which are supposed to take care of transforming of response. I don't think this code should have anything to do with apiserver code.